installing-packages.qmd

```diff
@@ -27,4 +27,69 @@ Installed --"library()"--> Loaded
 end
 ```
 
-[R Packages](https://r-pkgs.org/Structure.html) covers these phases of the package lifecycle in much more detail.
\ No newline at end of file
+[R Packages](https://r-pkgs.org/Structure.html) covers these phases of the package lifecycle in much more detail.
+
+## Binary packages
+
+- where to get them
+- how to know you got them
+
+##  Source packages
+
+The most common type of package you install is a **binary** package. Packages
+released on CRAN are built as pre-compiled binaries.
+
+However often it is useful to install packages which do not have a pre-built
+binary version available. This allows you to install development versions not
+yet released on CRAN, as well as older versions of released packages. It also lets
+you build your own packages locally.
+
+To install a source package you will need to [setup a development environment](#set-up-an-r-dev-environment).
+
+There are a few main functions used to install source packages.
+
+- `devtools::install_dev()` to install the latest development version of a CRAN package. ^[This will only work if the package includes a link to the development location in the package DESCRIPTION]
+- `devtools::install_github()` to install a package directly from GitHub, even if it is not on CRAN.
+- `devtools::install_version()` to install previously released CRAN versions of a package.
```

Comment:
The tidyverse/r-lib team is *this close* (imagine me pinching fingers together) to fully embracing pak for all of this. I think it's worth discussing whether to do so here.

That being said, when/if the above switch truly happens, we might actually change what powers these devtools functions, i.e. they might call out to pak instead of remotes.

Reply:
I am much further away from embracing `pak` but I think we could tabset it / `renv` as alternatives to `devtools`.

Reply:
Well, I think we have to contemplate that remotes (the package) is semi-deprecated and new energy is channeled into pak. I think the WTF audience is the sort of person who wants to be made at least aware of where things are heading and I think that's pak over remotes.

Do you have specific concerns that are relevant to WTF (vs. for discussion elsewhere)?

Reply:
I don't have any reservations about replacing references with `remotes` to `pak` and `renv`; I use the latter for all the workflows documented here.

Reply:
https://github.com/rstats-wtf/what-they-forgot/issues/94